parlai/core/torch_generator_agent.py

```diff
@@ -515,11 +524,19 @@ def eval_step(self, batch):
                 pad=self.NULL_IDX,
                 min_length=self.beam_min_length,
                 min_n_best=self.beam_min_n_best,
-                block_ngram=self.beam_block_ngram
+                block_ngram=self.beam_block_ngram,
+                num_iterations=self.beam_num_iterations,
+                min_distance=self.beam_min_distance
             )
             beam_preds_scores, _, beams = out
             preds, scores = zip(*beam_preds_scores)
 
+            if self.beam_num_iterations > 1:
+                # here the best candidate w.r.t. logprob is selected
+                # user can implement any other way to choose the best one
+                best_score_idx = torch.argmax(torch.Tensor(scores))
+                preds, scores = [preds[best_score_idx]], [scores[best_score_idx]]
```

Comment:
your line will not work since later it expects preds to be a *list* of predictions, thats why I put that list wrapping line 538. But thats also the effect of bsz 1 :)